### PR TITLE
Merge/june11

### PR DESCRIPTION
**Pull request by kirilklein**
## Summary by CodeRabbit

- **New Features**
	- Added automatic insertion of a predict token at the censor date during patient data processing.
	- Introduced new test scripts for validating trigger-outcome consistency and training an XGBoost baseline model.
- **Improvements**
	- Enhanced handling of vocabulary by loading it once and reusing it throughout data preparation.
	- Updated test cases to use more realistic patient event timing and age calculations.
	- Improved test coverage for censoring logic, including predict token insertion.
	- Added passing of predict token ID to censoring functions in causal data preparation.
	- Integrated data validation and baseline model training steps into the causal pipeline performance test workflow.
	- Improved robustness in selecting the latest performance file by datetime in test utilities.
	- Adjusted censoring and follow-up time parameters in causal finetuning configuration.
	- Refined regex patterns in filtering tests for better accuracy.
	- Enhanced pipeline scripts to clear old features and separate uncensored and censored finetuning steps with updated performance checks.
	- Expanded confidence interval range for lenient containment checks in estimation tests.
- **Bug Fixes**
	- Suppressed unnecessary warnings during regex-based filtering in data preparation.
- **Style**
	- Minor whitespace and import order adjustments for consistency.

### CODE REVIEW

File: corebehrt/functional/preparation/filter.py [hunk 1]
```diff
@@ -109,6 +119,22 @@ def censor_patient_with_delays(
...
+) -> PatientData:
+    """
+    Appends a predict token to the patient's data.
+    """
+    patient.concepts.append(predict_token_id)
+    patient.abspos.append(float(censor_date))
+    patient.segments.append(patient.segments[-1])
+    age_in_years = float((censor_date - patient.abspos[0]) / (365.25 * 24))
+    patient.ages.append(age_in_years)
     return patient
```
[the remark this answers is not in the document]
Comment (kirilklein) [reply corebehrt/functional/preparation/filter.py:138]: we always hav enon empty sequences, at least bg is always presnet
